vendor/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_image.go

```diff
@@ -169,7 +171,12 @@ func (o *ImageOptions) Run() error {
 				// Find the container to update, and update its image
 				for i, c := range spec.Containers {
 					if c.Name == name || name == "*" {
-						spec.Containers[i].Image = image
+						newImage, err := o.ResolveImage(image)
+						if err != nil {
+							allErrs = append(allErrs, fmt.Errorf("error: unable to set image for container %q: %v", name, err))
```

Comment:
@kargakis does it matter? it will be set anyway. (i will move it, no preference)